src/components/DataTable/tools/events.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Generic utility to compose event handlers so that consumers can supply their
+ * own event listeners on table components. The default heuristic here is to
+ * iterate through the given functions until `preventDefault` is called on the
+ * given event.
+ *
+ * @param {Array<Function>} fns array of functions to apply to the event
+ * @returns {Function}
+ */
+export const composeEventHandlers = fns => (event, ...args) => {
+  for (let i = 0; i < fns.length; i++) {
+    if (event.defaultPrevented) {
+      break;
+    }
+    if (typeof fns[i] === 'function') {
```

Comment:
Just a thought - Should we emit `invariant()` warning if `fns[i]` is not `undefined` but not a function?

Reply:
I definitely would not mind. Love introducing `invariant` for dev-time suggestions 😄